### PR TITLE
EWPP-3205: Various sync and mock related fixes.

### PR DESCRIPTION
**Pull request by upchuk**

### CODE REVIEW

File: modules/oe_translation_remote/tests/modules/oe_translation_remote_test/src/EventSubscriber/TranslationSourceEventSubscriber.php [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+  /**
+   * Reacts to the extract event.
+   *
+   * @param \Drupal\oe_translation\Event\TranslationSourceEvent $event
+   *   The event.
+   */
+  public function extract(TranslationSourceEvent $event): void {
+    $data = $event->getData();
+    $entity = $event->getEntity();
```
Comment (22Alexandra) [inline modules/oe_translation_remote/tests/modules/oe_translation_remote_test/src/EventSubscriber/TranslationSourceEventSubscriber.php]: It's not used.
```suggestion
```

**Review by 22Alexandra — DISMISSED** [review]

File: modules/oe_translation_remote/tests/src/FunctionalJavascript/RemoteTranslationTest.php [hunk 2]
```diff
@@ -715,4 +715,55 @@ public function testLanguageCheckboxesSelect(): void {
     $this->assertEquals('Select all', $checkbox->getParent()->find('css', 'label')->getText());
   }
 
+  /**
+   * Tests that the translation sync process throws a proper event subscriber.
```
Comment (hernani) [inline modules/oe_translation_remote/tests/src/FunctionalJavascript/RemoteTranslationTest.php]: Tests that the translation sync process throws an event which is handled by an event subscriber.

File: modules/oe_translation_remote/tests/src/FunctionalJavascript/RemoteTranslationTest.php [hunk 3]
```diff
@@ -715,4 +715,55 @@ public function testLanguageCheckboxesSelect(): void {
...
+    $this->getSession()->getPage()->pressButton('Save and send');
+    $this->assertSession()->pageTextContains('The translation request has been sent');
+
+    // Translate the request.
+    $requests = \Drupal::service('plugin.manager.oe_translation_remote.remote_translation_provider_manager')->getExistingTranslationRequests($node, TRUE);
+    $this->assertCount(1, $requests);
+    /** @var \Drupal\oe_translation_remote\TranslationRequestRemoteInterface $request */
+    $request = reset($requests);
+    TestRemoteTranslationMockHelper::translateRequest($request, 'fr');
+    // Remote from the translated data the #fake_value to mimic the fact that
```
Comment (hernani) [inline modules/oe_translation_remote/tests/src/FunctionalJavascript/RemoteTranslationTest.php]: Comment doesn't make sense

**Comment by upchuk** [comment]
Merging as it's approved by both @hernani and @22Alexandra

**Pull request merged**